src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1394,6 +1418,16 @@ private void setupCustomDeviceMapping(List<BlockDeviceMapping> deviceMappings) {
         }
     }
 
+    private boolean isUsingDynamicSshKeys() {
+        if (parent != null) {
+            String id = parent.getSshKeysCredentialsId();
+            if ((id == null) || (id.isEmpty())) {
```

Comment:
`Util.isEmpty` I guess, though best would be to make sure that the setter does not allow an empty string to be saved to begin with.

Reply:
AFAICT the setter is calling `fixEmpty()` so i will remove the additional check